src/main/java/com/myhome/controllers/CommunityAmenityController.java

```diff
@@ -57,6 +61,25 @@ public ResponseEntity deleteAmenity(@PathVariable String amenityId) {
     }
   }
 
+  @Operation(
+    description = "Update an amenity",
+    responses = {
+      @ApiResponse(responseCode = "204", description = "If updated successfully"),
+      @ApiResponse(responseCode = "400", description = "If amenity is not found"),
+    }
+  )
+  @PutMapping(path = "amenities/{amenityId}")
+  public ResponseEntity<Void> updateAmenity(@PathVariable String amenityId,
+                                            @Valid @RequestBody UpdateCommunityAmenityRequest request) {
+    boolean isUpdated = communityAmenitySDJpaService.updateAmenity(amenityId,
+      communityAmenityApiMapper.updateCommunityAmenityRequestToAmenityDto(request));
```

Comment:
Let's assign this `communityAmenityApiMapper.updateCommunityAmenityRequestToAmenityDto(request)` to a variable